cranelift/codegen/src/isa/x64/inst/mod.rs

```diff
@@ -1407,10 +1447,31 @@ fn x64_get_regs(inst: &Inst, collector: &mut RegUsageCollector) {
             src.get_regs_as_uses(collector);
             collector.add_mod(*dst);
         }
+        Inst::XmmRmRImm { src, dst, op, .. } => {
+            // In certain cases, instructions of this format can convert an XMM register into a
+            // define (e.g. an equality comparison); this extra logic is necessary to inform the
+            // registry allocator of a different register usage.
```

Comment:
I'd prefer to add a bit more detail here -- something like:

"In certain cases, instructions of this format can act as a definition of an XMM register, producing a value that is independent of its initial value. For example, a vector equality comparison (`cmppd` or `cmpps`) that compares a register to itself will generate all ones as a result, regardless of its value. From the register allocator's point of view, we should (i) record the first register, which is normally a mod, as a def instread; and (ii) not record the second register as a use, because it is the same as the first register (already handed)."

Maybe also "Any opcode for which `a OP b = constant` is true can be included here; we may add more in the future, such as `XOR`."?